web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -207,6 +223,40 @@ export const ConnectMyComputerContextProvider: FC<{
     }, [connectMyComputerService, rootClusterUri, workspacesService])
   );
 
+  const [removeAgentAttempt, removeAgent] = useAsync(
+    useCallback(async () => {
+      const [, error] = await killAgent();
+      if (error) {
+        throw error;
+      }
+      setCurrentActionKind('remove');
+
+      try {
+        await retryWithRelogin(ctx, rootClusterUri, () =>
+          ctx.connectMyComputerService.removeConnectMyComputerNode(
+            rootClusterUri
+          )
+        );
+      } catch (e) {
+        if (isAccessDeniedError(e)) {
```

Comment:
I was considering doing similar handling for "unauthenticated" errors, but idk if this needed at all.